boot/zephyr/flash_map_extended.c

```diff
@@ -135,3 +135,33 @@ __weak uint8_t flash_area_erased_val(const struct flash_area *fap)
     (void)fap;
     return ERASED_VAL;
 }
+
+int flash_area_get_sectors_fa(const struct flash_area *fa, uint32_t *count,
+    struct flash_sector *ret)
+{
+    off_t fa_off = fa->fa_off;
+    off_t offset = 0;
+    size_t fa_size = fa->fa_size;
```

Comment:
fa_off and fa_size only seem to be used for reading, can't the fa structure be used directly instead?

Reply:
Yes it can.